gen3-client/g3cmd/utils.go

```diff
@@ -27,19 +28,134 @@ type ManifestObject struct {
 	SubjectID string `json:"subject_id"`
 }
 
-// PresignedURLRequestObject represents the playload that sends to fence for getting a presignedURL for new object file
-type PresignedURLRequestObject struct {
+// InitRequestObject represents the payload that sends to fence for getting a singlepart upload presignedURL or init a multipart upload for new object file
+type InitRequestObject struct {
 	Filename string `json:"file_name"`
 }
 
+// MultipartUploadRequestObject represents the payload that sends to fence for getting a presignedURL for a part
+type MultipartUploadRequestObject struct {
+	Key        string `json:"key"`
+	UploadID   string `json:"uploadId"`
+	PartNumber int    `json:"partNumber"`
+}
+
+// MultipartCompleteRequestObject represents the payload that sends to fence for completeing a multipart upload
+type MultipartCompleteRequestObject struct {
+	Key      string                `json:"key"`
+	UploadID string                `json:"uploadId"`
+	Parts    []MultipartPartObject `json:"parts"`
+}
+
+// MultipartPartObject represents a part object
+type MultipartPartObject struct {
+	PartNumber int    `json:"PartNumber"`
+	ETag       string `json:"ETag"`
+}
+
 // FileInfo is a helper struct for including subdirname as filename
 type FileInfo struct {
 	FilePath string
 	Filename string
 }
 
-// FileSizeLimit is the maximun single file size (5GB)
-const FileSizeLimit = 5 * 1024 * 1024 * 1024
+const (
+	_ = iota
+	// KB is kilobytes
+	KB int64 = 1 << (10 * iota)
+	// MB is megabytes
+	MB
+	// GB is gigabytes
+	GB
+	// TB is terrabytes
+	TB
+)
+
+// FileSizeLimit is the maximun single file size for non-multipart upload (5GB)
+const FileSizeLimit = 5 * GB
+
+// MultipartFileSizeLimit is the maximun single file size for multipart upload (5TB)
+const MultipartFileSizeLimit = 5 * TB
+const maxMultipartNumber = 10000
+const minMultipartChunkSize = 5 * MB
+const defaultNumOfWorkers = 10
+
+// MaxRetryCount is the maximum retry number per record
+const MaxRetryCount = 5
+const maxWaitTime = 300
+
+// InitMultpartUpload helps sending requests to fence to init a multipart upload
+func InitMultpartUpload(uploadPath string, filePath string, includeSubDirName bool) (string, string, string, error) {
```

Comment:
```suggestion
func InitMultipartUpload(uploadPath string, filePath string, includeSubDirName bool) (string, string, string, error) {
```
typo in init, upload and complete function names

Reply:
🤦‍♂ fixed in https://github.com/uc-cdis/cdis-data-client/pull/50/commits/eda0b3c42dee0b4df5ede94e39baf196ed9aa941